h3/src/lib.rs

```diff
@@ -5,17 +5,25 @@
 pub mod client;
 pub mod error;
 pub mod quic;
+pub(crate) mod request;
 pub mod server;
 
 pub use error::Error;
+pub use proto::headers::Protocol;
 
 mod buf;
-mod connection;
-mod frame;
-mod proto;
+#[allow(missing_docs)]
```

Comment:
How much needs to be made public outside the crate? This makes a lot of the library internals public, which means refactors would be "breaking changes".

Reply:
I tried to only make public the bare minimal. 

users would have to go out of their way to enable the feature "allow_access_to_core" to enable access to all the private crates. 

it reminds me of rustls `dangerous_configuration` I can actually just call it dangerous config. 

https://github.com/rustls/rustls/blob/3d8b801221a6b4adfdca04934ecdda497423a0d8/rustls/Cargo.toml#L27

Reply:
The concern is that people can enable the feature and depend on the internals, and when we change things around in a refactor (such as in this PR), the public stuff changes and "breaks" them.

The "dangerous configuration" in rustls I don't think is meant to access unstable stuff, but rather to access crypto-security things. More like a "woah, are you sure".

Even if used just by `h3-webtransport`, if we make a change in `h3`, anyone depending on `h3-webtransport` who updates their `h3` dependency will see it "break".